sql/sql_admin.cc

```diff
@@ -534,6 +534,51 @@ static Check_result check_for_upgrade(THD *thd, dd::String_type &sname,
   return {false, result_code};
 }
 
+#ifdef WITH_WSREP
+/*
+   OPTIMIZE, REPAIR and ALTER may take MDL locks not only for the affected
+   table, but also for the table referenced by foreign key constraint.
+   ALTER additionally may take MTL lock of the tables that are referencing
+   altered table.
+   This wsrep_toi_replication() function handles TOI replication for OPTIMIZE
+   and REPAIR so that certification keys for potential FK parent tables are
+   also appended in the write set.
+   ALTER TABLE case is handled in alter table execution path.
+*/
+static bool wsrep_toi_replication(THD *thd, TABLE_LIST *tables) {
```

Comment:
Could we give a more specific name to this function?

Reply:
I would like to keep it as it is because this part comes from the bundle from Codership, so just to be more in sync...